### ISSUE TITLE
FunctionCallSignature works incorrectly

### ISSUE THREAD

**Issue by semirke** [PHPCSStandards/PHP_CodeSniffer#452]
## Describe the bug
Seems PEAR.Functions.FunctionCallSignature.Indent works incorrectly with some nesteted structures.
The sample will tell.

### Expected formatting
phpcs fires 
 5 | ERROR | [x] Multi-line function call not indented correctly; expected 4 spaces but found 8
   |       |     (PEAR.Functions.FunctionCallSignature.Indent)

```php

$fails = $this->req(
    "bs" => array(
        "type" => "number",
        "min" => 0,
        "default" => 0,
        "optional" => true,
    ),
);

```

### Phpcbf formatting 

```php
$fails = $this->req(
    "bs" => array(
    "type" => "number",
    "min" => 0,
    "default" => 0,
    "optional" => true,
    ),
```

### Sample from PEAR standard

From: https://pear.php.net/manual/en/standards.funcalls.php

```php
$this->someObject->subObject->callThisFunctionWithALongName(
    $this->someOtherFunc(
        $this->someEvenOtherFunc(
            'Help me!',
            array(
                'foo'  => 'bar',
                'spam' => 'eggs',
            ),
            23
        ),
        $this->someEvenOtherFunc()
    ),
    $this->wowowowowow(12)
);
```
### Custom ruleset
No custom ruleset 

### To reproduce
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```text

PHPCS output here
----------------------------------------------------------------------------------------------------------------------------------------------
FOUND 5 ERRORS AFFECTING 5 LINES
----------------------------------------------------------------------------------------------------------------------------------------------
 2 | ERROR | [ ] Missing file doc comment (PEAR.Commenting.FileComment.Missing)
 5 | ERROR | [x] Multi-line function call not indented correctly; expected 4 spaces but found 8
   |       |     (PEAR.Functions.FunctionCallSignature.Indent)
 6 | ERROR | [x] Multi-line function call not indented correctly; expected 4 spaces but found 8
   |       |     (PEAR.Functions.FunctionCallSignature.Indent)
 7 | ERROR | [x] Multi-line function call not indented correctly; expected 4 spaces but found 8
   |       |     (PEAR.Functions.FunctionCallSignature.Indent)
 8 | ERROR | [x] Multi-line function call not indented correctly; expected 4 spaces but found 8
   |       |     (PEAR.Functions.FunctionCallSignature.Indent)
----------------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 4 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------------------------------------------------------------------------------

Time: 41ms; Memory: 6MB

```

## Expected behavior

As above.

## Versions (please complete the following information)

|                                       |                                                                                                           |
|--------------------------|----------------------------------------------------------------------  |
| Operating System           | Debian 10                                                       |
| PHP version                    | 7.4                                                                                     |
| PHP_CodeSniffer version |PHP_CodeSniffer version 3.8.0 (stable) by Squiz (https://www.squiz.net)  |
| Standard                         |  PEAR |
| Install type                      | git clone |

## Please confirm

- [ x] I have searched the issue list and am not opening a duplicate issue.
- [ ] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [ x] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.

**Comment by semirke** [comment]
Hi, 
  I edited the issue and added more details with better sample code.

**Comment by semirke** [comment]
My guess is sniffer would need recursive processing of multi line function calls. Or from inside to outside.
I tried to look into it, but it was too much...
